### PR TITLE
Issue-15: Add support for Maybe#filter(Predicate)

### LINKED ISSUE #15: Add support for Maybe#Filter(Predicate)
* Passing the `predicate` returns the `Just` of the value.
* Failing the `predicate` or being a `Nothing` returns `Nothing`.

### PR DESCRIPTION
**Pull request by jlmorgan**
- [x] Squash commits before merging.

closes #15
SemVer:Minor
enhancement

### CODE REVIEW

File: src/FunctionalExtras/Data/Maybe.cs [hunk 1]
```diff
@@ -182,6 +195,17 @@ internal struct Just<A> : IMaybe<A>
     public bool Equals(IMaybe<A> other) => other is Just<A>
       && _value.Equals(((Just<A>) other)._value);
 
+    /// <summary>
+    /// Tests the underlying value against the {@code predicate}, returning the {@code Just} of the value for
```
Comment (doomchild) [inline src/FunctionalExtras/Data/Maybe.cs]: C# doesn't use Java's `{@code ... }` style.  These should be `<code>predicate</code>` and the like.

**Review by doomchild — CHANGES_REQUESTED** [review]
Documentation changes.

File: src/FunctionalExtras/Data/Maybe.cs (hunk 1 above)
Comment (jlmorgan) [reply src/FunctionalExtras/Data/Maybe.cs]: Fixed.

**Review by doomchild — APPROVED** [review]
Seems legit.

File: docs/Data/Maybe/instance/Filter.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+* `ArgumentNullException` if the `predicate` is `null`.
+
+## Examples
+
+```csharp
+Predicate<int> isEven = value => value % 2 == 0;
+Maybe.Just(0)
+  .Filter(isEven);
+// => Just(1)
```
Comment (JeremiahSanders) [inline docs/Data/Maybe/instance/Filter.md]: This example return doesn't match its input.

File: src/FunctionalExtras.Tests/Data/MaybeTests.cs [hunk 3]
```diff
@@ -55,6 +55,49 @@ public void ShouldReturnListOfNonNullValuesForMixedList()
...
+        {
+          IMaybe<int> testMaybe = Maybe.Just(0);
+          IMaybe<int> expectedResult = testMaybe;
+          IMaybe<int> actualResult = Maybe.Filter(testPredicate)(testMaybe);
+
+          Assert.Equal(expectedResult, actualResult);
+        }
+
+        [Fact]
+        public void ShouldReturnJustForFalsePredicate()
```
Comment (JeremiahSanders) [inline src/FunctionalExtras.Tests/Data/MaybeTests.cs]: `ShouldReturnJustForFalsePredicate` ➡ `ShouldReturnNothingForFalsePredicate`

File: docs/Data/Maybe/instance/Filter.md (hunk 2 above)
Comment (jlmorgan) [reply docs/Data/Maybe/instance/Filter.md]: Ooo… Good catch. (^_^)

File: src/FunctionalExtras.Tests/Data/MaybeTests.cs (hunk 3 above)
Comment (jlmorgan) [reply src/FunctionalExtras.Tests/Data/MaybeTests.cs]: Another good catch.

File: docs/Data/Maybe/instance/Filter.md (hunk 2 above)
Comment (jlmorgan) [reply docs/Data/Maybe/instance/Filter.md]: Fixed in `11d646d`

File: src/FunctionalExtras.Tests/Data/MaybeTests.cs (hunk 3 above)
Comment (jlmorgan) [reply src/FunctionalExtras.Tests/Data/MaybeTests.cs]: Fixed in `11d646d`.

**Review by JeremiahSanders — APPROVED** [review]

**Review by jlmorgan — COMMENTED** [review]
Ignore this.

**Comment by doomchild** [comment]
> Ignore this.

Okay, but not because you _told_ me to.

**Pull request merged**